bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/InstallationCleaner.java

```diff
@@ -45,28 +46,13 @@ class InstallationCleaner implements Runnable {
     }
 
     @Override
-    public void run() {
+    public synchronized void run() {
```

Comment:
I am missing something; this task is submitted by a `SingleThreadExecutor` in the BootableJar shutdown hook.
If we are marking it as synchronized, it only means that we could have more than one Bootable Jar instance for the same server home, right?
If that's true, the same marked file is also shared as a marked across the multiple Bootable JAar instances launched from the same home, wouldn't that be an issue at all?

Reply:
> this task is submitted by a SingleThreadExecutor in the BootableJar shutdown hook.
If we are marking it as synchronized, it only means that we could have more than one Bootable Jar instance for the same server home, right?

Well, even in that case, we are creating new instances of this InstallationCleaner on each shuftdown hook, so I don't get why the synchronized is required (or nice to have ) at this point.

Reply:
@yersan , we could have a timeout on the calling thread. The task (running in its own thread) is not yet done, and the calling thread will attempt to do a cleanup again. We need to synchronize at this point to avoid multiple cleanup in //. synchronize enforces it.

Reply:
@jfdenise ok, so it is not to allow dealing with muiltiple Bootable JARs from the same server home.

Ok, in that case, should not be InstallationCleaner.cleanup() method be the one that should be synchronized?

That's the method in common with the `InstallationCleaner.run()` and `InstallationCleaner.cleanupTimeout()` which are the entry points for the submitted task and the explicit `cleaner.cleanupTimeout()`

Reply:
@yersan The key piece is: Files.notExists(cleanupMarker)
We need to have all threads to share a common view on it. So all entry points to it should be synchronized.